src/link/spec_definition.rs

```diff
@@ -155,23 +155,19 @@ impl<T: SpecDefinition> SpecDefinitions<T> {
         }
     }
 
-    pub(crate) fn add(&mut self, definition: T) -> Result<(), FederationError> {
-        if *definition.identity() != self.identity {
-            return Err(SingleFederationError::Internal {
-                message: format!(
-                    "Cannot add definition for {} to the versions of definitions for {}",
-                    definition.to_string(),
-                    self.identity
-                ),
-            }
-            .into());
-        }
+    pub(crate) fn add(&mut self, definition: T) {
+        assert_eq!(
```

Comment:
since this will now panic, I'm wondering whether there would be any situation where we would want to still return partial schema even if this fails

Reply:
from what I can tell, this particular code is all set up by federation internally, never by users. The assertion can only fail if the `definition` has a different `identity` than its type suggests, which is a bug, and should be fixed by us so we have to find out as early and loudly as possible (ideally at compile time but a crash is the second best thing 😄 )